### PR TITLE
Change `username` to email

### LINKED ISSUE #236: Change username to email in code
Change every `username` occurence into `email`. It must include variables, function names, sequelize attribute.

Don't forget to migrate database. Ask @Embraser01 :)

This issue requires #228

### PR DESCRIPTION
**Pull request by kistora**
### Description of your PR

Change every `username` occurences to `email`. Will close #236 

### Pre-review TODO

- [x] PR is rebased on top of `origin/master` 
- [ ] PR provide new tests for new behaviors
- [x] PR doesn't introduce commented out code
- [x] PR has been tested locally on the main use cases
- [ ] Documentation have been updated if necessary

### CODE REVIEW

File: client/src/app/_services/auth.service.ts [hunk 1]
```diff
@@ -56,28 +56,28 @@ export class AuthService {
...
+      email,
       password,
       passwordToken,
       oldPassword,
     });
   }
 
-  verifyUsername(userId: number, username: string, password: string, usernameToken: string): Observable<any> {
-    return this.http.post('api/auth/username-verification', {
+  verifyEmail(userId: number, email: string, password: String, emailToken: string): Observable<any> {
```
Comment (Botweiser) [inline client/src/app/_services/auth.service.ts]: Code Smell Code Smell: Replace this 'String' wrapper object with primitive type 'string'. ([typescript:S1533](https://sonarcloud.io/organizations/k-fet/rules#rule_key=typescript%3AS1533))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=k-app&pullRequest=291&issues=AWU5BUYmOTG07BqbJp4b&open=AWU5BUYmOTG07BqbJp4b)

[ISSUE_KEY]: AWU5BUYmOTG07BqbJp4b

File: client/src/app/_services/auth.service.ts [hunk 2]
```diff
@@ -56,28 +56,28 @@ export class AuthService {
       .pipe(tap(this.clearUser.bind(this)));
   }
 
-  definePassword(username: string, password: string, passwordToken: string, oldPassword: string): Observable<any> {
+  definePassword(email: string, password: String, passwordToken: String, oldPassword: String): Observable<any> {
```
Comment (Botweiser) [inline client/src/app/_services/auth.service.ts]: Code Smell Code Smell: Replace this 'String' wrapper object with primitive type 'string'. ([typescript:S1533](https://sonarcloud.io/organizations/k-fet/rules#rule_key=typescript%3AS1533))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=k-app&pullRequest=291&issues=AWU5BUYmOTG07BqbJp4Y&open=AWU5BUYmOTG07BqbJp4Y)

[ISSUE_KEY]: AWU5BUYmOTG07BqbJp4Y

Comment (Botweiser) [inline client/src/app/_services/auth.service.ts]: Code Smell Code Smell: Replace this 'String' wrapper object with primitive type 'string'. ([typescript:S1533](https://sonarcloud.io/organizations/k-fet/rules#rule_key=typescript%3AS1533))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=k-app&pullRequest=291&issues=AWU5BUYmOTG07BqbJp4Z&open=AWU5BUYmOTG07BqbJp4Z)

[ISSUE_KEY]: AWU5BUYmOTG07BqbJp4Z

Comment (Botweiser) [inline client/src/app/_services/auth.service.ts]: Code Smell Code Smell: Replace this 'String' wrapper object with primitive type 'string'. ([typescript:S1533](https://sonarcloud.io/organizations/k-fet/rules#rule_key=typescript%3AS1533))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=k-app&pullRequest=291&issues=AWU5BUYmOTG07BqbJp4a&open=AWU5BUYmOTG07BqbJp4a)

[ISSUE_KEY]: AWU5BUYmOTG07BqbJp4a

File: client/src/app/app.module.ts [hunk 3]
```diff
@@ -9,7 +9,7 @@ import { FormsModule, ReactiveFormsModule } from '@angular/forms';
 
 // Libraries
 import { NgxPermissionsModule } from 'ngx-permissions';
-import { ReCaptchaModule } from 'angular2-recaptcha';
+import { ReCaptchaModule } from 'angular2-recaptcha/angular2-recaptcha';
```
Comment (Embraser01) [inline client/src/app/app.module.ts]: Why is it needed ?

File: server/app/controllers/me-controller.js [hunk 4]
```diff
@@ -33,6 +33,7 @@ async function me(req, res) {
  */
 async function updateMe(req, res) {
   const user = await authService.me(req.user.jit);
+  console.log(user.specialAccount);
```
Comment (Embraser01) [inline server/app/controllers/me-controller.js]: Remove console

File: client/src/app/app.module.ts (hunk 3 above)
Comment (kistora) [reply client/src/app/app.module.ts]: I don't know :)

File: server/app/controllers/me-controller.js (hunk 4 above)
Comment (kistora) [reply server/app/controllers/me-controller.js]: Yes

**Pull request merged**